tests/syslog/syslog_utils.py

```diff
@@ -0,0 +1,152 @@
+    if vrf:
+        cmd_add_syslog_server = "{} --vrf {} ".format(cmd_add_syslog_server, vrf)
+    if port:
+        cmd_add_syslog_server = "{} --port {} ".format(cmd_add_syslog_server, port)
+    logging.debug("add_syslog_server command is: {}".format(cmd_add_syslog_server))
+    return dut.command(cmd_add_syslog_server, module_ignore_errors=True)
+
+
+def del_syslog_server(dut, syslog_server_ip):
+    """
+    Del syslog server
+
+    Args:
+        dut (SonicHost): The target device
+        syslog_server_ip (str): Syslog server ip
+    """
+    dut.command('sudo config syslog del {} '.format(syslog_server_ip))
+
+
+def show_syslog(dut):
+    """
+    Show syslog config
+
+    Args:
+        dut (SonicHost): The target device
+    Return: Syslog config like below
+        [{
+            "server": "2.2.2.2",
+            "source": "1.1.1.1",
+            "port": "514",
+            "vrf": "default",
+          },
+          {
+            "server": "3.3.3.3",
+            "source": "4.4.4.4",
+            "port": "514",
+            "vrf": "mgmt",
+          },
+          ...
+        ]
+    """
+    return dut.show_and_parse('sudo show syslog')
+
+
+def create_vrf(dut, vrf):
+    """
+    Create Vrf
+
+    Args:
+        dut (SonicHost): The target device
+        vrf (str): vrf
+    """
+    dut.command('sudo config vrf add {} '.format(vrf), module_async=True)
+
+
+def remove_vrf(dut, vrf):
+    """
+    Remove Vrf
+
+    Args:
+        dut (SonicHost): The target device
+        vrf (str): vrf
+    """
+    return dut.command('sudo config vrf del {} '.format(vrf), module_ignore_errors=True)
+
+
+def bind_interface_to_vrf(dut, vrf, interface):
+    """
+    Bind interface to the specified vrf
+
+    Args:
+        dut (SonicHost): The target device
+        vrf (str): vrf
+        interface (str): interface
+    """
+    dut.command('sudo config interface vrf bind {} {} '.format(interface, vrf))
+
+
+def replace_ip_neigh(dut, neighbour, neigh_mac_addr, dev):
+    """
+    replace ip neigh
+
+    Args:
+        dut (SonicHost): The target device
+        neighbour (str): neighbour
+        neigh_mac_addr (str): neighbour mac address
+        dev (str): device
+
+    """
+    dut.command("sudo ip neigh replace {neighbor} lladdr {neigh_mac_addr} dev {dev}".format(
+        neighbor=neighbour,
+        neigh_mac_addr=neigh_mac_addr,
+        dev=dev))
+
+
+def capture_syslog_packets(dut, tcpdump_cmd):
+    """
+    Capture syslog packets
+
+    Args:
+        dut (SonicHost): The target device
+        tcpdump_cmd (str): tcpdump cmd
+    Return: filepath
+    """
+    logging.info("Start tcpdump: {}".format(tcpdump_cmd))
+
+    pcap_file_full_path = tcpdump_cmd.split("-w")[-1].strip()
+    dut.shell("sudo rm -f {}".format(pcap_file_full_path))
+    tcpdump_task, tcpdump_result = dut.shell(tcpdump_cmd, module_async=True)
+    # wait for starting tcpdump
+    time.sleep(5)
```

Comment:
>5 [](http://example.com/codeflow?start=15&length=1)

It will not work if the `tcpdump_cmd` completed < 5 seconds.

Reply:
This command of tcpdump_cmd executes for 30s, so it will not completed in 5s.
Waiting 5s  is to make sure that tcpdump is ready to capture packets, so reduce the noise.

Reply:
From single function point of view, it is not always true that "This command of tcpdump_cmd executes for 30s".

Reply:
I mean that the tcpdump_cmd (sudo timeout 30 tcpdump ...) has a timeout parameter, so it ensures the cmd can be executed for 30s.

Reply:
I understand the 5<30 should working for your current version. Since the constants are in 2 files, future dev may mistakenly modify one of them and break the assumption. So try get 5 from a function parameter, and let the caller could add some comments that 5<30 must be true.

Reply:
Fixed it. Please review.